daliuge-engine/dlg/drop.py

```diff
@@ -300,19 +300,19 @@ def getmembers(object, predicate=None):
         for attr_name, obj in getmembers(self, lambda a: not(inspect.isfunction(a) or isinstance(a, property))):
             if isinstance(obj, dlg_float_param):
                 value = kwargs.get(attr_name, obj.default_value)
-                if value is not None:
+                if value is not None and value is not '':
```

Comment:
`and value != ''`, otherwise it's an identity comparison. Likewise for the rest below.